lib/syskit/roby_app/plugin.rb

```diff
@@ -852,6 +858,37 @@ def self.setup_rest_interface(app, rest_api)
                 require 'syskit/roby_app/rest_api'
                 rest_api.mount REST_API => '/syskit'
             end
+
+            def self.start_local_log_transfer_server(tgt_dir, user, password, certificate)
+                @log_transfer_server = Syskit::RobyApp::LogTransferServer::SpawnServer.new(tgt_dir, user, password, certificate)
+            end
+
+            def create_self_signed_certificate
+                root_key = OpenSSL::PKey::RSA.new 2048 # the CA's public/private key
+                root_ca = OpenSSL::X509::Certificate.new
+                root_ca.version = 2 # cf. RFC 5280 - to make it a "v3" certificate
+                root_ca.serial = 1
+                root_ca.subject = OpenSSL::X509::Name.parse "/DC=org/DC=ruby-lang/CN=Ruby CA"
+                root_ca.issuer = root_ca.subject # root CA's are "self-signed"
+                root_ca.public_key = root_key.public_key
+                root_ca.not_before = Time.now
+                root_ca.not_after = root_ca.not_before + 2 * 365 * 24 * 60 * 60 # 2 years validity
```

Comment:
... make it a 100 years ... just to be sure. Or if you can leave it blank, that would be good too.

Note that it's *safe* to do so. We will be throwing it away on restart.